File: epi_judge_cpp/drawing_skyline.cc
#include <vector>
#include "test_framework/fmt_print.h"
#include "test_framework/generic_test.h"
#include "test_framework/serialization_traits.h"
using std::vector;

struct Rect {
  int left, right, height;
};
typedef vector<Rect> Skyline;

Skyline ComputeSkyline(const vector<Rect>& buildings) {
  // TODO - you fill in here.
  return {};
}
bool operator==(const Rect& a, const Rect& b) {
  return a.left == b.left && a.right == b.right && a.height == b.height;
}

template <>
struct SerializationTraits<Rect> : UserSerTraits<Rect, int, int, int> {};

std::ostream& operator<<(std::ostream& out, const Rect& r) {
  return PrintTo(out, std::make_tuple(r.left, r.right, r.height));
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"buildings"};
  return GenericTestMain(args, "drawing_skyline.cc", "drawing_skyline.tsv",
                         &ComputeSkyline, DefaultComparator{}, param_names);
}

File: epi_judge_python/drawing_skyline.py
import collections
import functools
from typing import List

from test_framework import generic_test
from test_framework.test_utils import enable_executor_hook

Rect = collections.namedtuple('Rect', ('left', 'right', 'height'))


def compute_skyline(buildings: List[Rect]) -> List[Rect]:
    # TODO - you fill in here.
    return []


@enable_executor_hook
def compute_skyline_wrapper(executor, buildings):
    buildings = [Rect(*x) for x in buildings]

    result = executor.run(functools.partial(compute_skyline, buildings))

    return [(x.left, x.right, x.height) for x in result]


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('drawing_skyline.py',
                                       'drawing_skyline.tsv',
                                       compute_skyline_wrapper))
